bin/main.rb

```diff
@@ -1,142 +1,107 @@
-    puts 'Please enter your names'
+  compare = Compare.new(player_1.name, player_2.name)
 
-  elsif player_1 == player_2 # if the player name is equal, the program should return
+  if compare.is_equal? && !compare.is_empty?
     puts 'Please enter a distinctive name'
 
-  elsif !player_1.empty? and !player_2.empty? # if the player_1 and player_2 is not empty,
-    puts 'happy gaming'
-    # break we break the loop
-    break
+  elsif compare.is_empty?
+    puts 'Please enter your names'
 
-  else # needs  both names
+  elsif compare.is_missing?
     puts 'Both names are required'
+
+  else
+    break
   end
+
 end
 
-# Assign either X or O to a player:
-# either random or always P1 = X, ect.
+first_player = Players.new(player_1.name, 'X', [])
+second_player = Players.new(player_2.name, 'O', [])
+
 sleep(1)
-puts "#{player_1} you are X"
-puts "#{player_2} you are 0"
+puts "#{player_1.name} you are #{first_player.sign}"
+puts "#{player_2.name} you are #{second_player.sign}"
 sleep(1)
-# countdown to the game:
+puts
 puts 'The game commences in:'
 puts 3
 sleep(1)
 puts 2
 sleep(1)
 puts 1
 sleep(1)
-# Create and Display the game board
-
-board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
 
-board_tokens = []
+board = Board.new
 
-board_1 = [] # rubocop:todo Naming/VariableNumber
+round = 1
 
-board_2 = [] # rubocop:todo Naming/VariableNumber
+winner = nil
 
-def game_board(board_tokens) # rubocop:todo Metrics/CyclomaticComplexity
-  puts " #{board_tokens[0] || ' '} | #{board_tokens[1] || ' '} | #{board_tokens[2] || ' '} "
-  puts '-----------'
-  puts " #{board_tokens[3] || ' '} | #{board_tokens[4] || ' '} | #{board_tokens[5] || ' '} "
-  puts '-----------'
-  puts " #{board_tokens[6] || ' '} | #{board_tokens[7] || ' '} | #{board_tokens[8] || ' '} "
-end
+current_player = first_player
 
-breakpoint = board.length
+loop do
+  puts
+  puts board.display
+  puts
 
-puts 'This is your game board.'
-puts 'The numbers represent the cell you can choose'
+  puts "Your turn #{current_player.name}"
 
-puts ' 1 | 2 | 3'
-puts '-----------'
-puts ' 4 | 5 | 6 '
-puts '-----------'
-puts ' 7 | 8 | 9 '
+  choice = gets.chomp.to_i
 
-sleep(2)
-
-until breakpoint.zero?
-
-  puts "#{player_1} it is your turn, please choose between #{board}"
-
-  loop do
-    choice_1 = gets.chomp.to_i # rubocop:todo Naming/VariableNumber
-    if board.include?(choice_1)
-      board_1 << choice_1
-      board.delete(choice_1)
-      board_1.each { |token| board_tokens[token - 1] = 'X' }
-      breakpoint -= 1
-
-      break
-    else
-      puts 'The selected cell is invalid!'
-      puts "please choose between #{board}"
-    end
+  while if board.is_valid?(choice)
+          board.update_board(choice - 1, current_player.sign)
+          current_player.choice << choice
+          break
+        else
+          puts 'please choose an available slot'
+        end
   end
 
-  sleep(1)
-
-  game_board(board_tokens)
+  is_winner = Winner.new(current_player.choice)
```

Comment:
This line indicates a wrong OOP design.
The current player choices you pass in parameters are used without any modification in the Winner class and moreover, the Winner class contains only one method checking the winner.

A better OOP implemention, according to your approach, is to merge the `Player` and `Winner` classes. 
You will no longer have to pass the current player's choices to any method. You will just do
```rb
current_player.real_win?
```